roadmap/implementers-guide/src/types/candidate.md

```diff
@@ -145,6 +145,9 @@ struct LocalValidationData {
 	/// which case the code upgrade should be applied at the end of the signaling
 	/// block.
 	code_upgrade_allowed: Option<BlockNumber>,
+
+	// TODO: Does this look sensible?
+	hrmp_ingress: BTreeMap<ParaId /* sender */, Vec<InboundHrmpMessage>>,
```

Comment:
Yeah, that seems fine. 

I would also expect a field or fields in either `GlobalValidationData` or `LocalValidationData` that can be used to ensure that not too many messages or too much data is sent.

Reply:
i guess we need such a field for upward messages too, but that seems beyond the scope of this PR

Reply:
> but that seems beyond the scope of this PR

https://github.com/paritytech/polkadot/issues/1534